### PR TITLE
Batch Norm implementation 

### PR DESCRIPTION
**Pull request by aksrajvanshi**
Addition: Implemented Batch Norm chapter of d2l book. (Section 7.5)

### CODE REVIEW

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 1]
```diff
@@ -0,0 +1,1020 @@
...
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "72"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "// utility class to store Batch Norm parameters such as Y, moving mean & moving Var.\n",
+    "class BatchNormObj {\n",
```
Comment (lanking520) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:320]: I would suggeet it called `BatchNormParam`. @frankfliu WDYT?

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 2]
```diff
@@ -0,0 +1,1020 @@
...
+    "\n",
+    "    private NDArray Y;\n",
```
Comment (lanking520) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:322]: Why we are keeping Y here, what did y do?

Comment (aksrajvanshi) [reply jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:322]: It's been used in the forward function (for returning Y) and also in the batchNormUpdate method. We need the other 2 as they are constantly changing too.

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 3]
```diff
@@ -0,0 +1,1020 @@
...
+    "        movingVar = manager.zeros(shape);\n",
+    "    }\n",
+    "\n",
+    "    @Override\n",
+    "    public String toString() {\n",
+    "        return \"BatchNormBlock()\";\n",
+    "    }\n",
+    "    \n",
+    "    @Override\n",
+    "    public NDList forward(ParameterStore parameterStore, NDList inputs, boolean training) {\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:489]: You can delete this forward because it is inherited from Block

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 4]
```diff
@@ -0,0 +1,1020 @@
...
+       ")"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "SequentialBlock net = new SequentialBlock();\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:567]: You can just do:
```
SequentialBlock net = new SequentialBlock()
  .add(...)
  ...
```

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 5]
```diff
@@ -0,0 +1,1020 @@
...
+   "outputs": [],
+   "source": [
+    "SequentialBlock block = new SequentialBlock();\n",
+    "\n",
+    "block\n",
+    "    .add(\n",
+    "            Conv2d.builder()\n",
+    "                    .setKernelShape(new Shape(5, 5))\n",
+    "                    .setFilters(6).build())\n",
+    "    .add(ai.djl.nn.norm.BatchNorm.builder().build())\n",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:814]: Why is this the full package path?

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb (hunk 2 above)
Comment (aksrajvanshi) [reply jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:322]: @lanking520 is this okay for you or I need to change anything with this ?

File: jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb [hunk 6]
```diff
@@ -0,0 +1,893 @@
...
+    "\n",
+    "System.out.printf(\"loss %.3f,\", trainLoss[numEpochs - 1]);\n",
+    "System.out.printf(\" train acc %.3f,\", trainAccuracy[numEpochs - 1]);\n",
+    "System.out.printf(\" test acc %.3f\\n\", testAccuracy[numEpochs - 1]);\n",
+    "System.out.printf(\"%.1f examples/sec\", trainIter.size() / (avgTrainTimePerEpoch / Math.pow(10, 9)));\n",
+    "System.out.println();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```
Comment (zachgk) [inline jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb:849]: Shouldn't this be a code cell?
